<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1515</width>
    <height>680</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: #303030; color: white;</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="pictureButtonLayout">
      <item>
       <layout class="QHBoxLayout" name="pictureLayout">
        <property name="sizeConstraint">
         <enum>QLayout::SetFixedSize</enum>
        </property>
        <item>
         <widget class="ui::InputLabel" name="inputLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>500</width>
            <height>500</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: #808080; color:black; border-radius: 10px;</string>
          </property>
          <property name="text">
           <string>拖动图片到此处</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="ui::OutputLabel" name="outputLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>500</width>
            <height>500</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: #808080; color:black; color:black; border-radius: 10px;</string>
          </property>
          <property name="text">
           <string>结果展示</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QGridLayout" name="buttonLayout" columnstretch="3,3,3,2">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <property name="leftMargin">
         <number>50</number>
        </property>
        <property name="topMargin">
         <number>10</number>
        </property>
        <property name="rightMargin">
         <number>50</number>
        </property>
        <property name="bottomMargin">
         <number>10</number>
        </property>
        <property name="spacing">
         <number>0</number>
        </property>
        <item row="1" column="1">
         <widget class="QPushButton" name="removeAllPointsButton">
          <property name="styleSheet">
           <string notr="true">background-color: #c0c0c0; color: black; margin: 5px 50px; height: 18px; border-radius: 3px;</string>
          </property>
          <property name="text">
           <string>清空所有点</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QComboBox" name="shapeComboBox">
          <property name="styleSheet">
           <string notr="true">QComboBox { background-color: #c0c0c0; color: black; margin: 5px 50px; height: 18px; border-radius: 3px; }
QComboBox::drop-down { subcontrol-origin: padding; subcontrol-position: right center; border-left-width: 1px; border-left-color: #a0a0a0; border-left-style: solid; }
QComboBox::down-arrow { width: 15px; height: 15px; }</string>
          </property>
          <item>
           <property name="text">
            <string>圆弧(三点)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>圆(三点)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>多段闭折线(多点)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>轮廓线(一点)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>直线(两点)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>线段(两点)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>平行四边形(四角)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>多段折线(多点)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>矩形(四角)</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="cameraButton">
          <property name="styleSheet">
           <string notr="true">background-color: #c0c0c0; color: black; margin: 5px 50px; height: 18px; border-radius: 3px;</string>
          </property>
          <property name="text">
           <string>摄像头</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QCheckBox" name="showContourCheckBox">
          <property name="styleSheet">
           <string notr="true">margin: 5px 50px; height: 18px;</string>
          </property>
          <property name="text">
           <string>显示轮廓</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="removeCurrentShapeButton">
          <property name="styleSheet">
           <string notr="true">background-color: #c0c0c0; color: black; margin: 5px 50px; height: 18px; border-radius: 3px;</string>
          </property>
          <property name="text">
           <string>删除当前图形</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="drawButton">
          <property name="styleSheet">
           <string notr="true">background-color: #c0c0c0; color: black; margin: 5px 50px; height: 18px; border-radius: 3px;</string>
          </property>
          <property name="text">
           <string>绘制图形</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QPushButton" name="resizeButton">
          <property name="styleSheet">
           <string notr="true">background-color: #c0c0c0; color: black; margin: 5px 50px; height: 18px; border-radius: 3px;</string>
          </property>
          <property name="text">
           <string>比例校正(单圆校正)</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="runButton">
          <property name="styleSheet">
           <string notr="true">background-color: #c0c0c0; color: black; margin: 5px 50px; height: 18px; border-radius: 3px;</string>
          </property>
          <property name="text">
           <string>运行</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="resizeLabel">
            <property name="styleSheet">
             <string notr="true">margin-left: 10px; height: 18px;</string>
            </property>
            <property name="text">
             <string>当前缩放比例:1.0</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="resizeDefaultButton">
            <property name="styleSheet">
             <string notr="true">background-color: #c0c0c0; color: black; margin-right: 20px; height: 18px; border-radius: 3px;</string>
            </property>
            <property name="text">
             <string>恢复默认</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="3">
         <widget class="QCheckBox" name="toNearestContourPointCheckBox">
          <property name="styleSheet">
           <string notr="true">margin: 5px 50px; height: 18px;</string>
          </property>
          <property name="text">
           <string>识别为最近轮廓点</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="3">
         <widget class="QCheckBox" name="showPointCheckBox">
          <property name="styleSheet">
           <string notr="true">margin: 5px 50px; height: 18px;</string>
          </property>
          <property name="text">
           <string>显示点</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="removeLastPointButton">
          <property name="styleSheet">
           <string notr="true">background-color: #c0c0c0; color: black; margin: 5px 50px; height: 18px; border-radius: 3px;</string>
          </property>
          <property name="text">
           <string>删除上个点</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QListWidget" name="shapeListWidget">
      <property name="minimumSize">
       <size>
        <width>256</width>
        <height>620</height>
       </size>
      </property>
      <property name="sizeIncrement">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: #808080; color:black; border-radius: 10px;</string>
      </property>
      <property name="resizeMode">
       <enum>QListView::Fixed</enum>
      </property>
      <property name="selectionRectVisible">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QGridLayout" name="detailLayout" columnstretch="5,5">
      <item row="0" column="1">
       <widget class="QLabel" name="value0Label">
        <property name="text">
         <string>参数值</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="name5Label">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="name7Label">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="value3Label">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="name1Label">
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="name6Label">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="name3Label">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="value1Label">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QLabel" name="value7Label">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLabel" name="value4Label">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="name0Label">
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="text">
         <string>参数名</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="name2Label">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="value2Label">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLabel" name="value5Label">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="name4Label">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QLabel" name="value6Label">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1515</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>选项</string>
    </property>
    <addaction name="saveResultPicture"/>
    <addaction name="saveShapeParams"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="saveResultPicture">
   <property name="text">
    <string>保存结果图片为</string>
   </property>
  </action>
  <action name="saveShapeParams">
   <property name="text">
    <string>保存图形参数至</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ui::InputLabel</class>
   <extends>QLabel</extends>
   <header>ui/InputLabel.hpp</header>
  </customwidget>
  <customwidget>
   <class>ui::OutputLabel</class>
   <extends>QLabel</extends>
   <header>ui/OutputLabel.hpp</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
